### DISCUSSION TITLE
Disable equality for one factory constructor

### DISCUSSION THREAD

**Discussion by AgPeHaJIuH1** [rrousselGit/freezed#981] [Q&A]
Can I disable equality generation for just one factory constructor?
I use flutter + bloc + freezed and I want one of the states (ErrorState) not to generate equality, so that bloc can react even to identical states and not ignore duplicate ones?
Now I have to do this as in the screenshot. Is there another way?

![image](https://github.com/rrousselGit/freezed/assets/29563929/e8b3015a-0ed3-4bae-a59c-444988f59ce9)

**Comment by AgPeHaJIuH1** [comment]
I would like equality to be generated for the "data" state, but not for the "success" and "failure" states.

**Comment by rrousselGit** [comment]
You can use `@With` to apply custom code on one specific case, and override the == to not compare properties

**Comment by AgPeHaJIuH1** [comment]
in the case of an abstract class, as in the documentation, a compilation error

![Screenshot_63](https://github.com/rrousselGit/freezed/assets/29563929/bcf20c3b-b1e4-4db1-bf93-08ebe5cf3e4c)
![Screenshot_64](https://github.com/rrousselGit/freezed/assets/29563929/aa914b7d-c2c4-4663-a842-a39eb21f1eb0)

in the case of a mixin, operators are not called

![Screenshot_65](https://github.com/rrousselGit/freezed/assets/29563929/e2686101-46ce-447b-bf46-eacf599918af)

what am I doing wrong? =(

**Comment by AgPeHaJIuH1** [comment]
It would be very convenient to be able to apply decorators not only to the class as a whole, but also to specific factory constructors. For example:

![image](https://github.com/rrousselGit/freezed/assets/29563929/de73ccd4-c93a-4760-9f98-db3f6cfc4cdc)
